docs/migration_guides/algorithms_migration.rst

```diff
@@ -0,0 +1,790 @@
+
+* `Opflow migration guide <https://qisk.it/opflow_migration>`_
+* `Quantum Instance migration guide <https://qisk.it/qi_migration>`_
+
+
+Choosing the Right Primitive for your Algorithm
+-----------------------------------------------
+*Back to* `TL;DR`_
+
+There are 3 different common configurations for algorithms that determine **which primitive import** you should
+be selecting:
+
+1. Running an algorithm with a statevector simulator
+   (ie. using :mod:`qiskit.opflow`\'s legacy :class:`.MatrixExpectation`):
+
+        - Reference Primitives (see `QAOA`_ example):
+
+        .. code-block:: python
+
+            from qiskit.primitives import Sampler, Estimator
+
+2. Running an algorithm using a "qasm" simulator/device (i.e. using :mod:`qiskit.opflow`\'s legacy :class:`.PauliExpectation`):
+
+        - Reference Primitives **with shots** (see `VQE`_ examples):
+
+        .. code-block:: python
+
+            from qiskit.primitives import Sampler, Estimator
+
+            sampler = Sampler(options={"shots": 100})
+            estimator = Estimator(options={"shots": 100})
+
+            # or...
+            sampler = Sampler()
+            job = sampler.run(circuits, shots=100)
+
+            estimator = Estimator()
+            job = estimator.run(circuits, observables, shots=100)
+
+        - Runtime Primitives (see `VQD`_ example):
+
+        .. code-block:: python
+
+            from qiskit_ibm_runtime import Sampler, Estimator
+
+        - Aer Primitives (see `VQE`_ examples):
+
+        .. code-block:: python
+
+            from qiskit_aer.primitives import Sampler, Estimator
+
+
+3. Running an algorithm on an Aer simulator using custom instruction (ie. using :mod:`qiskit.opflow`\'s legacy
+:class:`.AerPauliExpectation`):
+
+        - Aer Primitives with ``shots=None``, ``approximation=True`` (see `TrotterQRTE`_ example):
+
+        .. code-block:: python
+
+            from qiskit_aer.primitives import Sampler, Estimator
+
+            sampler = Sampler(run_options={"approximation": True, "shots": None})
+            estimator = Estimator(run_options={"approximation": True, "shots": None})
+
+
+.. note::
+
+    In general, in order to know which primitive to use instead of :class:`~qiskit.utils.QuantumInstance`,
+    you should ask yourself two questions:
+
+    1. What is the minimal unit of information used by your algorithm?
+        a. **Expectation value** - you will need an ``Estimator``
+        b. **Probability distribution** (from sampling the device) - you will need a ``Sampler``
+
+    2. How do you want to execute your circuits?
+        a. Using **local** statevector simulators for quick prototyping: **Reference Primitives**
+        b. Using **local** noisy simulations for finer algorithm tuning: **Aer Primitives**
+        c. Accessing **runtime-enabled backends** (or cloud simulators): **Runtime Primitives**
+        d. Accessing **non runtime-enabled backends** : **Backend Primitives**
```

Comment:
For a you can also choose the Aer primitives. For the VQE code I was doing recently it was faster running overall though it does depend on what you are doing.
The Runtime primitives are more the IBM Runtime. I would say d. is more if your provider does not have native primitives you can use a Backend primitive and pass a backend from it.